akka-serialization-jackson/src/main/scala/akka/serialization/jackson/JacksonSerializer.scala

```diff
@@ -0,0 +1,369 @@
+   * serializer when serializing (`toBinary`). For deserialization (`fromBinary`) the serializer-id is
+   * used for selecting serializer.
+   * Here we use `serialization-bindings` also and more importantly when deserializing (fromBinary)
+   * to check that the manifest class is of a known (registered) type.
+   * The drawback of using `serialization-bindings` for this is that an old class can't be removed
+   * from `serialization-bindings` when it's not used for serialization but still used for
+   * deserialization (e.g. rolling update with serialization changes). It's also
+   * not possible to change a binding from a JacksonSerializer to another serializer (e.g. protobuf)
+   * and still bind with the same class (interface).
+   * If this is too limiting we can add another config property as an additional way to
+   * whitelist classes that are not bound to this serializer with serialization-bindings.
+   */
+  private def isInWhitelist(clazz: Class[_]): Boolean = {
+    try {
+      // The reason for using isInstanceOf rather than `eq this` is to allow change of
+      // serializizer within the Jackson family, but we don't trust other serializers
+      // because they might be bound to open-ended interfaces like java.io.Serializable.
+      serialization.serializerFor(clazz).isInstanceOf[JacksonSerializer]
+    } catch {
+      case NonFatal(_) => false // not bound
+    }
+  }
+
+  /**
+   * Check that serialization-bindings are not configured with open-ended interfaces,
+   * like java.lang.Object, bound to this serializer.
+   *
+   * This check is run on first access since it can't be run from constructor because SerializationExtension
+   * can't be accessed from there.
+   */
+  private def checkAllowedSerializationBindings(): Unit = {
+    if (!serializationBindingsCheckedOk) {
+      def isBindingOk(clazz: Class[_]): Boolean =
+        try {
+          serialization.serializerFor(clazz) ne this
+        } catch {
+          case NonFatal(_) => true // not bound
+        }
+
+      JacksonSerializer.disallowedSerializationBindings.foreach { clazz =>
+        if (!isBindingOk(clazz)) {
+          val warnMsg = "For security reasons it's not allowed to bind open-ended interfaces like " +
+            s"[${clazz.getName}] to [${getClass.getName}]. " +
+            "Change your akka.actor.serialization-bindings configuration."
+          log.warning(LogMarker.Security, warnMsg)
+          throw new IllegalArgumentException(warnMsg)
+        }
+      }
+      serializationBindingsCheckedOk = true
+    }
+  }
+
+  private def parseManifest(manifest: String) = {
+    val i = manifest.lastIndexOf('#')
+    val fromVersion = if (i == -1) 1 else manifest.substring(i + 1).toInt
+    val manifestClassName = if (i == -1) manifest else manifest.substring(0, i)
+    (fromVersion, manifestClassName)
+  }
+
+  def compress(bytes: Array[Byte]): Array[Byte] = {
+    val bos = new ByteArrayOutputStream(BufferSize)
```

Comment:
That's simple enough but before committing to a compression scheme we might want to consider if there's an alternative to gzip (e.g. LZ4).